homeassistant/helpers/entity_component.py

```diff
@@ -364,11 +364,13 @@ def _update_entity_states(self, now):
 
         This method must be run in the event loop.
         """
-        if self._process_updates:
+        if self._process_updates.locked():
+            self.component.logger.warning(
+                "Updating %s took longer than the scheduled update "
+                "interval %s", self.component.domain, self.scan_interval)
```

Comment:
We should not use `self.component.domain` because it only tells us "light" for example. I want to be able to name the platform so that we can point fingers. "Updating **hue** light took longer…" (means we have to pass the platform name into the constructor)